MediaBrowser.Common/Cryptography/PasswordHash.cs

```diff
@@ -30,6 +31,16 @@ public PasswordHash(string id, byte[] hash, byte[] salt)
 
         public PasswordHash(string id, byte[] hash, byte[] salt, Dictionary<string, string> parameters)
         {
+            if (id == null)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (id.Length == 0)
+            {
+                throw new ArgumentException("String can't be empty", nameof(id));
+            }
```

Comment:
Why not `string.IsNullOrEmpty`?

Reply:
We already checked `null`

Reply:
Right, but it could just be a call to `IsNullOrEmpty` instead of both of these checks

Reply:
Different exceptions

Reply:
Is there a benefit to doing that? the string is invalid either way, and the rest of the code is written the way I described. If this is the way we want to go, that's fine, but we should be consistent about it.

Reply:
More expressive, consistent with the .Net BCL?